public/javascripts/songButton.js

```diff
@@ -0,0 +1,146 @@
+var addOrDeleteSong = function(id) {
```

Comment:
Put these functions into a namespace to avoid conflicts. For example:
`function songButton () {
  return {
   addOrDeleteSong: function (){}
  }
}`

Furthermore, it's generally not a good idea to combine add and delete into the same operation like this. Let each function do one thing.